runtime/tests/runtime_integration_tests/test_invoke_dpe.rs

```diff
@@ -297,3 +298,69 @@ fn test_invoke_dpe_rotate_context() {
 
     assert!(rotate_ctx_resp.handle.is_default());
 }
+
+fn check_dice_extension_criticality(cert: &[u8], expected_criticality: bool) {
+    let mut parser = X509CertificateParser::new().with_deep_parse_extensions(true);
+    let Ok((_, cert)) = parser
+            .parse(cert) else {
+                panic!("Could not parse x509 certificate from CertifyKey!");
+            };
+    for extension in cert.iter_extensions() {
+        // Unknown extensions are DICE extensions, and they should match the
+        // criticality set by the DPE instance.
+        if extension.parsed_extension().unsupported() {
+            assert_eq!(extension.critical, expected_criticality);
+        }
+    }
+}
+
+#[test]
+fn test_invoke_dpe_certify_key_with_critical_dice_extensions() {
```

Comment:
nit: Shouldn't these test names be `with_non_critical_dice_extensions`?

Reply:
Whoops, Yes!

Thanks!